doc/classes/Resource.xml

```diff
@@ -77,6 +77,7 @@
 		</member>
 		<member name="resource_name" type="String" setter="set_name" getter="get_name" default="&quot;&quot;">
 			An optional name for this resource. When defined, its value is displayed to represent the resource in the Inspector dock. For built-in scripts, the name is displayed as part of the tab name in the script editor.
+			[b]Note:[/b] Some resource types do not support this property and will always give value [code]""[/code].
```

Comment:
```suggestion
			[b]Note:[/b] Some resource formats do not support resource names, so this property can be reset after reload.
```

The same resource type can be saved in different formats.

Reply:
ok